src/hls-audio-track.js

```diff
@@ -0,0 +1,114 @@
+import {AudioTrack} from 'video.js';
+import PlaylistLoader from './playlist-loader';
+
+/**
+ * HlsAudioTrack extends video.js audio tracks but adds HLS
+ * specific data storage such as playlist loaders, mediaGroups
+ * and default/autoselect
+ *
+ * @param {Object} options options to create HlsAudioTrack with
+ * @class HlsAudioTrack
+ * @extends AudioTrack
+ */
+export default class HlsAudioTrack extends AudioTrack {
+  constructor(options) {
+    super({
+      kind: options.default ? 'main' : 'alternative',
+      enabled: options.default || false,
+      language: options.language,
+      label: options.label
+    });
+
+    let trackProps = {
+      autoselect: options.autoselect || false,
+      default: options.default || false,
+      withCredentials: options.withCredentials || false
+    };
+
+    // set noop
+    let set = () => {};
+
+    Object.defineProperties(this, {
```

Comment:
I understand your point about consistency but I think it's more important to use the least-powerful tool that gets the job done. Custom properties imply extra functionality to anyone debugging and that would be misleading in this case.